end-user-application-machines/new-agent-vm/Install-Idle-Shutdown.sh

```diff
@@ -1,257 +1,271 @@
 # there are instances of pcoip-server processes running.
 
-## Global settings
-##
-# minimum number of minutes machine can be idle before shutting down
-MIN_IDLE_WAIT_TIME=5
-
-# maximum number of minutes machine can be idle before shutting down
-MAX_IDLE_WAIT_TIME=10000 # about 7 days
-
-# minimum number of minutes to wait between polling machine session status
-MIN_POLL_INTERVAL=1
-
-# maximum number of minutes to wait between polling machine session status
-MAX_POLL_INTERVAL=60
-##
-
-####################################################################
-# Log to Syslog
-####################################################################
-function logit {
-   logger --id $$ -t "ShutdownIdleAgent" "$*"
-}
-
-####################################################################
-# If there are active remote ssh connections or we detect PCoIP return 1, else 0
-# Note that a remote ssh session is one where the remote address is NOT "0.0.0.0" (ipv4
-# or "::0" or "::" (ipv6)
-####################################################################
-function activeConnectionsExist() {
-    set -x
-
-    # first, check to see if anyone is connected remotely
-    #  last will produce something like the list below,
-
-    #  someuser  pts/2         tervdiw10dev13.t Wed Jul 12 10:51   still logged in
-    #  someuser  pts/1         0.0.0.0          Wed Jul 12 10:36   gone - no logout
-    #  someuser  pts/0         0.0.0.0          Wed Jul 12 10:35   gone - no logout
-    #  someuser  :100          0.0.0.0          Wed Jul 12 10:35   gone - no logout
-    #  reboot   system reboot  0.0.0.0          Wed Jul 12 10:35   gone - no logout
-    #  
-    #  wtmp begins Wed Jul 12 10:35:17 2017
-
-    # in this case, pts/2 has a remotely logged on user, but the rest are local ssh connections we
-    # can ignore, including the system one.  Format is slightly different when the system status is displayed
-        
-    ipList=$(/usr/bin/last  | /usr/bin/grep 'still logged in' | /usr/bin/awk '{ if (!($0~"wtmp begins")) { if ($2 == "system") {print $4} else {print $3 }}}')
-
-    for i in $ipList; do
-        if [ $i != "0.0.0.0" ] && [ $i != "::" ] && [ $i != "::0" ]; then
-            # a remote ssh connection detected
-            echo 1
-            return
-        fi
-    done
-
-    # Is PCoIP active?
-    /bin/ps -e | /bin/grep -s pcoip-server  > /dev/null 2>&1
-    rc=$?
-
-    if [ $rc -eq 0 ]; then
-        # someone is connected via pcoip
-        echo 1
-        return
-    fi
-
-    echo 0
+syslog.openlog("ShutdownIdleAgent", syslog.LOG_PID)
+
+# Used for measuring CPU Usage, initialize to 0
+BusyTime_last = BusyTime_now = IdleTime_now = IdleTime_last = 0
+
+# File containing state information
+startTimeFile = "/tmp/CAMIdleState"
+cpuUsageFile = "/tmp/CAMIdleStateCPU"
+
+def getStartTime():
+    """
+    Check what the start time is
+    """
+    if os.path.exists(startTimeFile):
+        f = open(startTimeFile, 'rb')
```

Comment:
It's a good idea to use a context block for file I/O, like this:
```python
with open(startTimeFile, 'rb') as f:
    startTime = f.read()
```
The context block takes care of the underlying file handle management and keeps things neat (i.e. flushing when you write, ensuring that it's closed when the block exits regardless of whether errors occur). :)